ocaml/typing/typedtree.mli

```diff
@@ -398,10 +398,14 @@ and comprehension_iterator =
 and 'k case =
     {
      c_lhs: 'k general_pattern;
-     c_guard: expression option;
+     c_guard: guard option;
      c_rhs: expression;
     }
 
+and guard =
+  | Predicate of expression
+  | Pattern of expression * Layouts.sort * computation general_pattern
```

Comment:
Whose sort is this? If it's the scrutinee's sort (the same as `Texp_match`), we should say so in a comment. More generally, I think this type declaration is missing a comment.

Reply:
added a comment